tests/multimodel/test_fitzhugh_nagumo.py

```diff
@@ -0,0 +1,176 @@
+        return np.vstack([system.integrate(time) for time in times])
+
+
+class TestFitzHughNagumoMass(MassTestCase):
+    def _create_mass(self):
+        fhn = FitzHughNagumoMass()
+        fhn.index = 0
+        fhn.idx_state_var = 0
+        fhn.init_mass()
+        return fhn
+
+    def test_init(self):
+        fhn = self._create_mass()
+        self.assertTrue(isinstance(fhn, FitzHughNagumoMass))
+        self.assertDictEqual(fhn.params, DEFAULT_PARAMS)
+        coupling_variables = {k: 0.0 for k in fhn.required_couplings}
+        self.assertEqual(len(fhn._derivatives(coupling_variables)), fhn.num_state_variables)
+        self.assertEqual(len(fhn.initial_state), fhn.num_state_variables)
+        self.assertEqual(len(fhn.noise_input_idx), fhn.num_noise_variables)
+
+    def test_run(self):
+        fhn = self._create_mass()
+        result = self._run_mass(fhn, DURATION, DT)
+        self.assertTrue(isinstance(result, np.ndarray))
+        self.assertTupleEqual(result.shape, (int(DURATION / DT), fhn.num_state_variables))
+
+
+class TestFitzHughNagumoNetworkNode(unittest.TestCase):
+    def _create_node(self):
+        node = FitzHughNagumoNetworkNode(seed=SEED)
+        node.index = 0
+        node.idx_state_var = 0
+        node.init_node()
+        return node
+
+    def test_init(self):
+        fhn = self._create_node()
+        self.assertTrue(isinstance(fhn, FitzHughNagumoNetworkNode))
+        self.assertEqual(len(fhn), 1)
+        self.assertDictEqual(fhn[0].params, DEFAULT_PARAMS)
+        self.assertEqual(len(fhn.default_network_coupling), 2)
+        np.testing.assert_equal(np.array(fhn[0].initial_state), fhn.initial_state)
+
+    def test_run(self):
+        fhn = self._create_node()
+        all_results = []
+        for backend, noise_func in BACKENDS_TO_TEST.items():
+            result = fhn.run(
+                DURATION, DT, noise_func(ZeroInput(DURATION, DT, fhn.num_noise_variables)), backend=backend,
+            )
+            self.assertTrue(isinstance(result, xr.Dataset))
+            self.assertEqual(len(result), fhn.num_state_variables)
+            self.assertTrue(all(state_var in result for state_var in fhn.state_variable_names[0]))
+            self.assertTrue(
+                all(result[state_var].shape == (int(DURATION / DT), 1) for state_var in fhn.state_variable_names[0])
+            )
+            all_results.append(result)
+        # test results are the same from different backends
+        for state_var in all_results[0]:
+            corr_mat = np.corrcoef(
+                np.vstack([result[state_var].values.flatten().astype(float) for result in all_results])
+            )
+            self.assertTrue(np.greater(corr_mat, CORR_THRESHOLD).all())
+
+    def test_compare_w_neurolib_native_model(self):
+        """
+        Compare with neurolib's native FitzHugh-Nagumo model.
+        """
+        # run this model
+        fhn_multi = self._create_node()
```

Comment:
here run against native `neurolib`'s implementation is tested (if available, hence not for Wong-Wang)

Reply:
TODO: Implement WW soon :)

Reply:
Ehm, have you tested it with noise actually?